packages/monorepo/src/components/nx-configurator.ts

```diff
@@ -383,18 +401,56 @@ export class NxConfigurator extends Component implements INxProjectCore {
             "@nx/devkit": "^16",
           },
           private: true,
-          workspaces: this.project.subprojects
+          engines: {
+            node: ">=16",
+            pnpm: ">=8",
+          },
+          scripts: this.project.tasks.all
+            .filter((t) => t.name !== "install")
+            .reduce(
+              (p, c) => ({
+                ...p,
+                [c.name]: NodePackageUtils.command.projen(
+                  NodePackageManager.PNPM,
+                  c.name
+                ),
+              }),
+              {}
```

Comment:
Nit: prefer `Object.fromEntries(this.project.tasks.all.filter(...).map((c) => [c.name, NodePackageUtils.command...]))` :)